Content.Server/Chat/Managers/ChatManager.cs

```diff
@@ -79,16 +88,63 @@ public void DispatchServerAnnouncement(string message, Color? colorOverride = nu
+            bool playDefaultSound = true, Color? colorOverride = null)
         {
             var messageWrap = Loc.GetString("chat-manager-sender-announcement-wrap-message", ("sender", sender));
-            ChatMessageToAll(ChatChannel.Radio, message, messageWrap, colorOverride);
+            ChatMessageToAll(ChatChannel.Radio, message, messageWrap);
             if (playDefaultSound)
             {
                 SoundSystem.Play(Filter.Broadcast(), "/Audio/Announcements/announce.ogg", AudioParams.Default.WithVolume(-2f));
             }
+            _adminLogger.Add(LogType.Chat, LogImpact.Low, $"Global station announcement from {sender}: {message}");
+        }
+
+        /// <summary>
+        /// Dispatches an announcement on a specific station
+        /// </summary>
+        /// <param name="source">The entity making the announcement (used to determine the station)</param>
+        /// <param name="message">The contents of the message</param>
+        /// <param name="sender">The sender (Communications Console in Communications Console Announcement)</param>
+        /// <param name="playDefaultSound">Play the announcement sound</param>
+        /// <param name="colorOverride">Optional color for the announcement message</param>
+        public void DispatchStationAnnouncement(EntityUid source, string message, string sender = "Central Command", bool playDefaultSound = true, Color? colorOverride = null)
+        {
+            var messageWrap = Loc.GetString("chat-manager-sender-announcement-wrap-message", ("sender", sender));
 
-            _adminLogger.Add(LogType.Chat, LogImpact.Low, $"Station Announcement from {sender}: {message}");
+            var station = _stationSystem.GetOwningStation(source);
+            ChatMessageToAll(ChatChannel.Radio, message, messageWrap, colorOverride);
+            if (station != null)
+            {
+                if(!_entityManager.TryGetComponent<StationDataComponent>(station, out var stationDataComp)) return;
+                foreach (var gridEnt in stationDataComp.Grids)
+                {
+                    var filter = Filter.BroadcastGrid(gridEnt);
+                    ChatMessageToManyFiltered(filter, ChatChannel.Radio, message, messageWrap, source, true, colorOverride);
+                    if (playDefaultSound)
+                    {
+                        SoundSystem.Play(filter, "/Audio/Announcements/announce.ogg", AudioParams.Default.WithVolume(-2f));
+                    }
+                }
+            }
+            else
+            {
+                var filter = Filter.Pvs(source);
+                ChatMessageToManyFiltered(filter, ChatChannel.Radio, message, messageWrap, source, true, colorOverride);
+                if (playDefaultSound)
+                {
+                    SoundSystem.Play(filter, "/Audio/Announcements/announce.ogg", AudioParams.Default.WithVolume(-2f));
+                }
```

Comment:
Should these even be on the manager and not the system? They're all sim dependent, they even add the IEntityManager dependency.

Reply:
It was kind of like this when I started. Mirror said they forgot to move it so :shrug:

Reply:
uhhh should probably be moved then? I don't imagine it'd be too painful. ChatManager should really only be used for OOC I think.